### PR TITLE
Sublime Text Settings suggestions

### PR DESCRIPTION
**Pull request by deathaxe**
Here are some more suggestions for the settings module

### CODE REVIEW

File: settings.py [hunk 1]
```diff
@@ -116,7 +116,13 @@ def int_or_float(string):
 
 
 def _settings():
-    return sublime.load_settings("PackageDev.sublime-settings")
+    """Return global package settings object or load it first if required."""
```
Comment (FichteFoll) [inline settings.py]: FYI, ST caches settings in its internal API already, which is why calling `.load_settings` repeatedly is acceptable, but I'm okay with the change. The objects were still being created for no real benefit.

```py
>>> sublime.load_settings("PackageDev.sublime-settings").settings_id == sublime.load_settings("PackageDev.sublime-settings").settings_id
True
```

Comment (deathaxe) [inline settings.py]: > ST caches settings in its internal API already

I know. That's why I found creating a new python object on each call somehow a waste. This is what I meant with creating "garbage" as the objects need to be collected by the garbage collector again and again. I recently found this way to cache the object in one of @r-steins implementations and found it a quite interesting construct.

**Comment by FichteFoll** [comment]
Regarding quotation marks, I have the following guidelines:

- Generally, use double quotes. This is prevalently for arbitrary "text", like messages printed or constructed file paths.
- Use single quotes for symbol-like purposes, such as dictionary keys (or setting keys), function parameters (e.g. `open(mode='r')` or `str.decode('utf-8')`). Basically, situations in where the exact content of the string matters.
- Additionally, use single quotes if you'd have to escape double quotes within the string.

I find this guideline to be a good balance act between arbitrarily choosing either as the default and adding semantic meaning to a string based on the quotes used.

I try to follow these for my personal use cases, but I don't necessary go around changing other people's contributed code to adhere to them, which is also why I only changed the quotation marks for strings that I edited anyway in that commit.

**Comment by FichteFoll** [comment]
Cherry-picked all but the quotation commit.

**Pull request closed without merging**